lib/openstack_query/runners/query_runner.py

```diff
@@ -45,18 +48,54 @@ def run(
         """
 
         with self._connection_cls(cloud_account.name.lower()) as conn:
-            resource_objects = (
-                self._parse_subset(conn, from_subset)
-                if from_subset
-                else self._run_query(conn, server_side_filters, **kwargs)
-            )
+            if from_subset:
+                resource_objects = self._parse_subset(conn, from_subset)
+            else:
+                query_meta_params = self._parse_meta_params(conn, **kwargs)
+                resource_objects = self._run_query(
+                    conn, server_side_filters, **query_meta_params
+                )
 
         if client_side_filter_func and not server_side_filters:
             resource_objects = self._apply_client_side_filter(
                 resource_objects, client_side_filter_func
             )
         return resource_objects
 
+    def _run_paginated_query(
+        self,
+        paginated_call: Callable,
+        server_side_filters: Optional[ServerSideFilters] = None,
+    ):
+        """
+        Helper method for running a query using pagination - openstacksdk calls usually return a maximum number of
+        values - (set by limit) and to continue getting values we can pass a "marker" value of the last item seen to
+        continue the query - this speeds up the time needed to run queries
+        :param paginated_call: A lambda function which takes a openstacksdk call which allows limit and marker to be set
+        :param server_side_filters: A set of filters to pass to openstacksdk call
+        """
+
+        paginated_filters = {"limit": self._LIMIT_FOR_PAGINATION, "marker": None}
+        paginated_filters.update(server_side_filters)
+        query_res = []
+
+        curr_marker = None
+        while True:
+            for i, server in enumerate(paginated_call(**paginated_filters)):
+                query_res.append(server)
+                # openstacksdk calls break after going over pagination limit
+                if i == self._LIMIT_FOR_PAGINATION - 1:
+                    # restart the for loop with marker set
+                    paginated_filters.update({"marker": server["id"]})
```

Comment:
There's a reference to a server id here, is this meant to be just resource id instead?